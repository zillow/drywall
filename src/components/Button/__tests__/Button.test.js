import React from 'react';
import TestRenderer from 'react-test-renderer';
import { isStyledComponent } from 'styled-components';
import { Button } from '../../../index';

describe('<Button>', () => {
    it('can render without theme', () => {
        const testRenderer = TestRenderer.create(<Button>Test button</Button>);
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    it('is a styled-component', () => {
        expect(isStyledComponent(Button)).toBe(true);
    });
    it('supports the as prop', () => {
        const testRenderer = TestRenderer.create(<Button as="a">Test button</Button>);
        const tree = testRenderer.toJSON();
        expect(tree.type).toBe('a');
        expect(tree).toMatchSnapshot();
    });
});
