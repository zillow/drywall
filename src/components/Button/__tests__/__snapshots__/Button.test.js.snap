// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> can render without theme 1`] = `
<button
  className="Button-sc-1gypvbx-0 "
>
  Test button
</button>
`;

exports[`<Button> supports the as prop 1`] = `
<a
  className="Button-sc-1gypvbx-0 "
>
  Test button
</a>
`;
